### ISSUE TITLE
Notepad++ appx package interferes with sysprep

### ISSUE THREAD

**Issue by levicki** [notepad-plus-plus/notepad-plus-plus#13775]
While preparing custom Windows 11 image (based on ISO downloaded from Microsoft VS subscription) in audit mode I installed Notepad++ 8.5.2.

After I finished with other customization tasks I tried running `sysprep` to generalize the image and return it to OOBE state which failed with the following error:
![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/16415478/8934f806-33e6-43eb-9e72-6f66abc08474)

When I checked with `Get-AppxPackage *notepad*` command from PowerShell I got this:
![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/16415478/44e1e748-4272-46ff-aeec-5fedbbbb35e7)

The thing is, I did not install that package &mdash; I only ran regular 8.5.2 installer from the official website. I removed it and `sysprep` worked normally afterwards. Later I tried installing 8.5.3 update, and the APPX package was reinstalled so I had to remove it again.

My questions:

1. If NPP installer is not installing it, where is it coming from?
2. If NPP installer is installing it, why is it doing that covertly?
3. If NPP installer will be installing it in the future, will there be the option to not install it?
4. If no option, then would it at least be provisioned for all users so that `sysprep` doesn't fail?

Please advise.

**Comment by donho** [comment]
Related to https://github.com/notepad-plus-plus/nppShell/issues/29

**Comment by Neustradamus** [comment]
To follow

**Comment by levicki** [comment]
@Neustradamus 

> To follow

1. Are you aware that **everyone** subscribed to these 3 issues you replied to was just notified of your spam reply 3 times?
2. Don't you know there is a giant Subscribe button you can click if you want to follow an issue?

![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/16415478/bcd285d0-f5d1-4fd2-a3e4-b0abe1a8bce1)

**Comment by dhs-rec** [comment]
While it wasn't a problem anymore for some time, I'm now seeing it again, since at least version 8.9 (and possibly a couple of versions before), on Windows Server 2025. The problem is that `sysprep` simply seems to hang. OTOH, the problem does NOT appear on Windows Server 2022.

**Comment by dhs-rec** [comment]
Did a few test runs, and found that 8.8.6 is the last version that doesn't break sysprep on Server 2025.

**Comment by willard-t-johnson** [comment]
> Did a few test runs, and found that 8.8.6 is the last version that doesn't break sysprep on Server 2025.

Here to confirm this when attempting to generalize W11 as well.

**Comment by sky-flix** [comment]
> > Did a few test runs, and found that 8.8.6 is the last version that doesn't break sysprep on Server 2025.
> 
> Here to confirm this when attempting to generalize W11 as well.

You can simply remove the appx-package before running your sysprep using the following powershell command (as admin)
remove-appxpackage -allusers -package NotepadPlusPlus_1.0.0.0_neutral__2247w0b46hfww

The application will still be installed and accessible and will re-configure the appx package on first use.

**Comment by willard-t-johnson** [comment]
> > > Did a few test runs, and found that 8.8.6 is the last version that doesn't break sysprep on Server 2025.
> > 
> > 
> > Here to confirm this when attempting to generalize W11 as well.
> 
> You can simply remove the appx-package before running your sysprep using the following powershell command (as admin) remove-appxpackage -allusers -package NotepadPlusPlus_1.0.0.0_neutral__2247w0b46hfww
> 
> The application will still be installed and accessible and will re-configure the appx package on first use.

Cheers, appreciate you.

**Comment by dhs-rec** [comment]
@sky-flix,

> The application will still be installed and accessible and will re-configure the appx package on first use.

Sure, but that's just a workaround. It wasn't needed up to 8.8.6, so it's a regression introduced in 8.8.7 that should be fixed.

**Comment by Limerick-gh** [comment]
> 
> You can simply remove the appx-package before running your sysprep using the following powershell command (as admin) remove-appxpackage -allusers -package NotepadPlusPlus_1.0.0.0_neutral__2247w0b46hfww
> 

I assume this would be necessary every time the template and Notepad++ is being updated, before running sysprep?

**Comment by csdesign3** [comment]
The problem persists... even on 8.9.3 cannot do sysprep..

**Comment by csit-support** [comment]
Problem still persists, guess it has no priority so far.
Sadly we had to move on now to a different editor, due to the nerves lost in the sysprep process. I hope this will eventually be resolved as it's still one of the best editors to use

**Comment by xomx** [comment]
> The problem persists... even on 8.9.3 cannot do sysprep..

> Problem still persists, guess it has no priority so far.

The solution is described in the linked NppShell issue above:
1. before the syspreping, use `Get-AppxPackage -Name *NotepadPlusPlus* | Reset-AppxPackage`
2. in the meantime,  don't use right-clicking in Explorer (otherwise the NppShell sparse package self-registers again)
3. sysprep

**Comment by dhs-rec** [comment]
@xomx, no, that's not a solution, it's a workaround. This is a regression that has been introduced with 8.8.7 and should thus be fixed.

**Comment by xomx** [comment]
> no, that's not a solution, it's a workaround.

Sure. If you insist, a workaround then.

I'm not the NppShell author but from I know, the problem is in using of the MS sparse package way, which has some substantial limitations (in comparison with the full msix sln). So the current situation is a compromise in between having a functional Win11+ context menu + some sysprep discomfort, and having only the older Win10- context menu but sysprep ok.


> This is a regression that has been introduced with 8.8.7 and should thus be fixed.

The only relevant NppShell change, in-between the N++ v8.8.6 (NppShell v1.5)  and 8.8.7 (NppShell v1.6), has been switching from the expired DigiCert code signing certificate to the Globalsign one (at that time, the author of the N++ project finally managed to register the NOTEPAD++ association for the purposes of the N++ project, so N++ could obtain a legitimate OV cert), so currently I have no idea what can be fixed there ... :

https://github.com/notepad-plus-plus/nppShell/commits/master/

<img width="1022" height="763" alt="Image" src="https://github.com/user-attachments/assets/a97c2a9d-3029-4399-bb11-1ac757e4ded3" />

**Comment by dhs-rec** [comment]
> Sure. If you insist, a workaround then.

Yes, I do, because the problem hits us during the automated creation of Amazon Machine Images. The workaround is not automatable.
